File: delivery-service/db/src/main/java/org/delivery/db/userOrderMenu/UserOrderMenuRepository.java
package org.delivery.db.userOrderMenu;

import org.delivery.db.userOrderMenu.enums.UserOrderMenuStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface UserOrderMenuRepository extends JpaRepository<UserOrderMenuEntity, Long> {

    List<UserOrderMenuEntity> findAllByUserOrderIdAndStatus(Long userOrderId, UserOrderMenuStatus status);
}

File: delivery-service/db/src/main/kotlin/org/delivery/db/userOrderMenu/UserOrderMenuRepository.kt
package org.delivery.db.userOrderMenu

import org.delivery.db.userOrderMenu.enums.UserOrderMenuStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserOrderMenuRepository : JpaRepository<UserOrderMenuEntity, Long> {
    fun findAllByUserOrderIdAndStatus(userOrderId: Long?, status: UserOrderMenuStatus?): List<UserOrderMenuEntity>
}
